src/adc_readout/integration_test/FPGASim.cpp

```diff
@@ -157,7 +157,7 @@ void FPGASim::addSamplingRun(void const *const DataPtr, size_t Bytes,
   auto BufferSizes = StandbyBuffer->getBufferSizes();
   if (not Success or BufferSizes.second - BufferSizes.first < 20) {
     transmitStandbyBuffer();
-    assert(StandbyBuffer->addSamplingRun(DataPtr, Bytes, CurrentRefTimeNS));
+    StandbyBuffer->addSamplingRun(DataPtr, Bytes, CurrentRefTimeNS);
```

Comment:
That's a pretty big change, but I can see it makes sense!
If we are playing around with clang-tidy, it has this cool check https://clang.llvm.org/extra/clang-tidy/checks/bugprone-assert-side-effect.html